### ISSUE TITLE
How to show Fireworks on custom dialog ?

### ISSUE THREAD

**Issue by MrLoveQD** [plattysoft/Leonids#39]
Hi.
Thanks you so much for this library.
It's great and ease to use.

I'm using ParticleSystem to show fireworks, and i want show it on custom  dialog.
I'm trying show it at a view on custom dialog, but it show wrong position.

How to i can ?

**Comment by plattysoft** [comment]
I haven't tried that. but the system uses the window coordinates to get the X and Y of the View, so "in theory" should work fine.

Sorry I can't help you better.

**Comment by MrLoveQD** [comment]
Oh, thanks you.
It don't work for me.

**Comment by lvaizer** [comment]
wont work on dialogs
+1

**Comment by Aystub** [comment]
I have a custom AppCompatDialog I've built and in it I have a FrameLayout that's the length of the dialog and 200dp high (basically a rectangle). At the top of the dialog I have an emitter that's the length of the dialog. So I have confetti rain down inside my FrameLayout basically. I got it to work with the library so here's a simplified version of my dialog class:

```
public class BadgeDialog extends AppCompatDialog {

    public BadgeDialog(Context ctx){ 
        super(context);
        // Need to set the OwnerActivity so library can reference it
        if (context instanceof Activity) {
            setOwnerActivity((Activity) context);
        }
        init();
     }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    private void init() {
        setContentView(R.layout.dialog_badge);
        FrameLayout awardHolder = (FrameLayout) findViewById(R.id.award_holder);    
        View emitter = (View) findViewById(R.id.emitter);

        if(getOwnerActivity() != null) {
            // Make sure the emitter View has been measured before attaching the library to it
            ViewTreeObserver vto = emitter.getViewTreeObserver();
            vto.addOnGlobalLayoutListener(new ViewTreeObserver.OnGlobalLayoutListener() {
                @Override
                public void onGlobalLayout() {
                    emitter.getViewTreeObserver().removeGlobalOnLayoutListener(this);
                    new ParticleSystem(getOwnerActivity(), 100, R.drawable.confeti2, 3000)
                            .setAcceleration(0.00013f, 90)
                            .setSpeedByComponentsRange(0f, 0f, 0.05f, 0.1f)
                            .setFadeOut(200, new AccelerateInterpolator())
                            .setParentViewGroup(awardHolder)
                            .emitWithGravity(emitter, Gravity.BOTTOM, 30);

                    new ParticleSystem(getOwnerActivity(), 100, R.drawable.confeti3, 3000)
                            .setAcceleration(0.00013f, 90)
                            .setSpeedByComponentsRange(0f, 0f, 0.05f, 0.1f)
                            .setFadeOut(200, new AccelerateInterpolator())
                            .setParentViewGroup(awardHolder)
                            .emitWithGravity(emitter, Gravity.BOTTOM, 30);
                }
            });
        }
    }
}
```

**Comment by plattysoft** [comment]
this should have been fixed with #58
